rcl/src/rcl/node_resolve_name.c

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl/node.h"
+
+#include "rcutils/error_handling.h"
+#include "rcutils/logging_macros.h"
+#include "rcutils/types/string_map.h"
+
+#include "rmw/error_handling.h"
+#include "rmw/validate_full_topic_name.h"
+
+#include "rcl/error_handling.h"
+#include "rcl/expand_topic_name.h"
+#include "rcl/remap.h"
+
+#include "./remap_impl.h"
+
+static
+rcl_ret_t
+rcl_resolve_name(
+  const rcl_arguments_t * local_args,
+  const rcl_arguments_t * global_args,
+  const char * input_topic_name,
+  const char * node_name,
+  const char * node_namespace,
+  rcl_allocator_t allocator,
+  bool is_service,
+  bool only_expand,
+  char ** output_topic_name)
+{
+  // the other arguments are checked by rcl_expand_topic_name() and rcl_remap_topic_name()
```

Comment:
I guess you mean `rcl_remap_name()`.
```suggestion
  // the other arguments are checked by rcl_expand_topic_name() and rcl_remap_name()
```